### ISSUE TITLE
Make DAVDroid integration more seamless

### ISSUE THREAD

**Issue by MorrisJobke** [nextcloud/android#626]
* clicking the "setup sync" button it should:
   * automatically generate an app password and doesn't ask the user to type in the password in DAVDroid again
   * automatically set the users email address instead of asking the user for it (even if prefilled)
   * by default sync all calendars and address books and don't require the user to tick all those items manually

@tobiasKaminsky @AndyScherzinger Is this possible? This would improve the UX a lot 😃 

cc @mario

**Comment by MorrisJobke** [comment]
cc @oparoz

**Comment by jancborchardt** [comment]
Is it even possible to trigger the download more automatically if you don't have it? Like not redirect to the Play store but automatically show the popover to accept? Just a thought.

Otherwise totally agree with Morris! :+1:

**Comment by AndyScherzinger** [comment]
> Is it even possible to trigger the download more automatically if you don't have it? Like not redirect to the Play store but automatically show the popover to accept? Just a thought.

That is (technically/API-wise) not possible, we can only redirect to the store itself

**Comment by mario** [comment]
Installing the package SHOULD be possible starting from Lollipop (DEVICE OWNER permission @AndyScherzinger), but since Davdroid is a paid app that becomes an issue.

**Comment by AndyScherzinger** [comment]
So that is a no then, right?

**Comment by mario** [comment]
Indeed @AndyScherzinger :)

**Comment by AndyScherzinger** [comment]
@MorrisJobke at the moment we fill username/server-url, where should th eemail be placed and where would I find the tick boxes? None of this can be done atm sine it would require a change on the DAVDroid side first, so cc @rfc2822

**Comment by MorrisJobke** [comment]
> where should the email be placed and where would I find the tick boxes?

The email input appears on the next page (when you click "login") and the tick boxes aren't visible there. The app then closes and goes back to the nextcloud app with a little notification "Sync successfully set up" (or something like this). But it isn't - you need to open DAVDroid again. Go into the account and check the calendars and address books you want to sync.

**Comment by strugee** [comment]
@AndyScherzinger I'm not 100% sure but I _believe_ in the case where the app is installed via F-Droid, you should be able to download the F-Droid APK and prompt the user to install it, and F-Droid will install upgrades in the future because it has an F-Droid signature. That might be too much complexity, though, since I'm guessing Google Play downloads account for the majority.

**Comment by tobiasKaminsky** [comment]
Although I would love to have a more seamless way, I guess it is currently not that easy to implement and it is a one-time setup as this does not change...

**Comment by AndyScherzinger** [comment]
True, nevertheless the original request is not about making installation more seamless but the setup of DAVDroid, so let's focus on that part first :)

**Comment by tobiasKaminsky** [comment]
Then we should discuss this with DAVdroid guys, right? Or can we do from our app-side something?

**Comment by AndyScherzinger** [comment]
That is why I mentioned @rfc2822 :)

**Comment by rfc2822** [comment]
Would be cool if you could create a thread in our DAVdroid forums, because I rarely check my Github account.

Maybe it would be possible that after DAVdroid account creation, the DAVdroid account activity is launched (either by DAVdroid or NextCloud app), so that users can edit their subscriptions and settings.

**Comment by AndyScherzinger** [comment]
No, not yet

**Comment by jancborchardt** [comment]
Any update here @AndyScherzinger @mario @tobiasKaminsky? Would really be cool to have better integration here. :)

**Comment by AndyScherzinger** [comment]
No yet, nobody started working on this, simply a capacity matter that will prevent any work on this topic.

**Comment by mario** [comment]
Also we now have contacts backup that is tightly integrated into the app.

**Comment by joshtrichards** [comment]
Related more recent discussion: #4691
